src/data/p5.TypedDict.js

```diff
@@ -0,0 +1,371 @@
+};
+
+// /**
+//  * Creates a new p5.NumberDict using the key-value pair that you pass in
+//  *
+//  * @method createNumberDict()
+//  * @param {}
+//  * @param {}
+//  * @return {p5.NumberDict}
+//  * @example
+//  * <div>
+//  * <code>
+//  * </code>
+//  * </div>
+//  *
+//  * @alt
+//  * ALT TEXT HERE EVENTUALLY
+//  */
+p5.prototype.createNumberDict = function() {
+  return new p5.NumberDict(arguments[0], arguments[1]);
+};
+
+p5.TypedDict = function() {
+};
+
+/**
+ * Returns the number of key-value pairs currently in Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.size = function(){
+  return this.count;
+};
+
+/**
+ * Returns true if dictionary has key value pair
+ *
+ *
+ */
+p5.TypedDict.prototype.hasKey = function(key) {
+  return this.data.hasOwnProperty(key);
+};
+
+/**
+ * Returns the key-value pair located at key
+ *
+ *
+ */
+p5.TypedDict.prototype.get = function(key) {
+  if(this.data.hasOwnProperty(key)){
+    return this.data[key];
+  }else{
+    console.log(key + ' does not exist in this Dictionary');
+  }
+};
+
+/**
+ * Changes the value of key if in Dictionary otherwise makes new one
+ *
+ *
+ */
+p5.TypedDict.prototype.set = function(key, value) {
+  if (arguments.length === 2) {
+    if(!this.data.hasOwnProperty(key)){
+      this.create(key, value);
+    } else {
+      this.data[key] = value;
+    }
+  }
+};
+
+
+p5.TypedDict.prototype._addObj = function(obj) {
+  for (var key in obj) {
+    if(this._validate(obj[key])) {
+      this.data[key] = obj[key];
+      this.count++;
+    }
+  }
+};
+
+
+/**
+ * Creates a new key-value pair in Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.create = function() {
+  if(arguments.length === 1) {
+    if(arguments[0] instanceof Object) {
+      this._addObj(arguments[0]);
+    }
+  }
+  else if(arguments.length === 2) {
+    var key = arguments[0];
+    var value = arguments[1];
+    if (this._validate(value)) {
+      this.data[key] = value;
+      this.count++;
+    } else {
+      console.log('those values dont work for this dictionary type');
+    }
+  } else {
+    console.log('In order to create a new Dictionary entry you must pass ' +
+      'an object or a key, value pair');
+  }
+};
+
+/**
+ * Empties out Dictionary object
+ *
+ *
+ */
+p5.TypedDict.prototype.clear = function(){
+  for(var key in this.data) {
+    delete this.data[key];
+  }
+  this.count = 0;
+};
+
+/**
+ * Removes a key-value pair in the Dictionary
+ *
+ *
+ */
+p5.TypedDict.prototype.remove = function(key) {
+  if(this.data.hasOwnProperty(key)) {
+    delete this.data[key];
+    this.count--;
+  } else {
+    throw key + ' does not exist in this Dictionary';
+  }
+};
+
+
+/**
+ * Logs the list of items currently in the Dictionary
+ *
+ *
+ */
+p5.TypedDict.prototype.print = function() {
+  for (var item in this.data) {
+    console.log('key:' + item + ' value:' + this.data[item]);
+  }
+};
+
+p5.TypedDict.prototype._validate = function(key, value) {
+  return true;
```

Comment:
maybe the default method could make use of the type of object.. we already have number or string in the name, instead of having to write a custom validation method for each.

for example, on construction we could store a type:
`this.type = m.constructor.name.replace('Dict', '').toLowerCase();`

then the test would be
`return (typeof value === this.type);`

something like that?